src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -5826,6 +5826,7 @@ bool StorageReplicatedMergeTree::executeMetadataAlter(const StorageReplicatedMer
     Coordination::Requests requests;
     requests.emplace_back(zkutil::makeSetRequest(fs::path(replica_path) / "columns", entry.columns_str, -1));
     requests.emplace_back(zkutil::makeSetRequest(fs::path(replica_path) / "metadata", entry.metadata_str, -1));
+    requests.emplace_back(zkutil::makeSetRequest(fs::path(replica_path) / "metadata_version", std::to_string(entry.alter_version), -1));
```

Comment:
Need more explanation about this change.

Reply:
We want to keep `metadata_version` in memory always consistent with `metadata_verion` in ZooKeeper. Before it was not done in single transaction, that modifies `metadata` and `columns` because `metadata_version` node was not always created. Now we can safely update `metadata_version` in single transaction.
I think it does not make sense to write this as comment in code, because it will only complicate code read, it is clear why we want to modify all replica metadata in single transaction.